cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -1962,15 +1962,50 @@ INSTANTIATE_TEST_SUITE_P(AllNull, TestTableSortIndicesRandom,
 // ----------------------------------------------------------------------
 // Tests for Rank
 
-void AssertRank(const std::shared_ptr<Array>& input, SortOrder order,
-                NullPlacement null_placement, RankOptions::Tiebreaker tiebreaker,
+struct RankTestCase {
```

Comment:
Hmm... sorry to bother you with this, but it would be more idiomatic to use a GTest fixture (i.e. a class derived from `::testing::Test`).

You could have something like:
```c++
class TestRank : public ::testing::Test {
 public:
  void SetInput(const std::shared_ptr<Array>& array) {
    ...
```

Reply:
Yeah I agree. That would unify the base test names too (as of right now, it's kind of a pain to filter out the rank tests as a set).